src/GeositeFramework/js/Plugin.js

```diff
@@ -446,6 +460,56 @@ require(['use!Geosite',
             model.set('$uiContainer', $uiContainer);
         }
 
+        function setupPrintableMap(pluginObject, $printSandbox, previewDeferred) {
+            var mapMarkup = N.app.templates['template-map-preview']({ pluginName: pluginObject.toolbarName }),
+                $mapPrint = $($.trim(mapMarkup)),
+                $printPreview = $('#print-preview-sandbox'),
+                mapReadyDeferred = $.Deferred(),
+                mapHeight = pluginObject.previewMapSize[1],
+                mapWidth = pluginObject.previewMapSize[0];
+
+            // If the plugin is not set up for map print preview, don't set up a map
+            // and resolve any pending print-preview map operations
+            if (!pluginObject.usePrintPreviewMap) {
+                previewDeferred.resolve();
+                mapReadyDeferred.resolve();
+                return mapReadyDeferred;
+            }
+
+            // Setup a print-preview window for the user to select an extent and zoom level
+            // that will be persisted at print due to its fixed size.
+            TINY.box.show({
+                animate: false,
+                html: $mapPrint[0].outerHTML,
+                boxid: 'print-preview-container',
+                width: _.max([mapWidth, 500]),
+                fixed: true,
+                maskopacity: 40,
+                openjs: function () {
+                    // Remove any calculated size so that it contains the full map
+                    $('#print-preview-container').css({ height: 'initial', width: 'initial' });
+
+                    // Set the supplied height & width on the map
+                    $('#plugin-print-preview-map').css({ height: mapHeight, width: mapWidth });
+
+                    var map = new esri.Map('plugin-print-preview-map', { basemap: 'topo' });
+
+                    mapReadyDeferred.resolve(map);
+
+                    $('#print-preview-print').on('click', function() {
+                        // Move the map from the print preview dialog to the sandbox where
+                        // the plugin can mess with it's positioning among its other elements
+                        $(map.container).detach().appendTo($printSandbox);
+                        TINY.box.hide();
+                        $printPreview.hide();
+                        previewDeferred.resolve();
+                    });
+                }
+            });
+
+            return mapReadyDeferred;
```

Comment:
The `esriMap` isn't able to be constructed until the container element is on the DOM, which doesn't happen until `openjs` executes, hence the need for this additional deferred.  It's resolved when the map is constructed and is passed the map as an argument.